### PR TITLE
feat: add file logging

### PR DESCRIPTION
**Pull request by dni**
logs into `lnbits.log` with info loglevel and `debug.log` with DEBUG loglevel files will be saved into `LNBITS_DATA_DIR`.

retentions is configurable but defaults to `30 days`

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -8,6 +8,10 @@ PORT=5000
 
 DEBUG=false
 
+# logging into files inside LNBITS_DATA_FOLDER
+LOG_FILE=true
+LOG_RETENTION="30 days"
```
Comment (motorina0) [inline .env.example:13]: - let make it 90 days, storage is cheap

Comment (motorina0) [reply .env.example:13]:
```suggestion
LOG_RETENTION_DAYS=30
```
- users might have a typo, use upper case, miss adding a space (`30days`), etc. Better to just have a number value.

File: lnbits/app.py [hunk 2]
```diff
@@ -526,6 +526,21 @@ def configure_logger() -> None:
     log_level: str = "DEBUG" if settings.debug else "INFO"
     formatter = Formatter()
     logger.add(sys.stdout, level=log_level, format=formatter.format)
+
+    if settings.log_file:
+        logger.add(
+            Path(settings.lnbits_data_folder, "lnbits.log"),
```
Comment (motorina0) [inline lnbits/app.py:532]:
```suggestion
Path(settings.lnbits_data_folder, "logs", "lnbits.log"),
```
- do not store all log files on the top level

File: lnbits/app.py [hunk 3]
```diff
@@ -526,6 +526,21 @@ def configure_logger() -> None:
...
+            retention=settings.log_retention,
```
Comment (motorina0) [inline lnbits/app.py:533]:
```suggestion
            retention=f"{settings.log_retention} days",
```

File: lnbits/settings.py [hunk 4]
```diff
@@ -291,6 +291,8 @@ class EnvSettings(LNbitsSettings):
     lnbits_extensions_path: str = Field(default="lnbits")
     super_user: str = Field(default="")
     version: str = Field(default="0.0.0")
+    log_file: bool = Field(default=True)
+    log_retention: str = Field(default="30 days")
```
Comment (motorina0) [inline lnbits/settings.py:295]:
```suggestion
    log_retention: int = Field(default="90")
```

File: .env.example [hunk 5]
```diff
@@ -8,6 +8,10 @@ PORT=5000
 
 DEBUG=false
 
+# logging into files inside LNBITS_DATA_FOLDER
+LOG_FILE=true
```
Comment (motorina0) [inline .env.example:12]:
```suggestion
ENABLE_LOG_TO_FILE=true
```
- `LOG_FILE` var name suggests that it represents the name of the log file

File: lnbits/settings.py [hunk 6]
```diff
@@ -291,6 +291,8 @@ class EnvSettings(LNbitsSettings):
     lnbits_extensions_path: str = Field(default="lnbits")
     super_user: str = Field(default="")
     version: str = Field(default="0.0.0")
+    log_file: bool = Field(default=True)
```
Comment (motorina0) [inline lnbits/settings.py:294]:
```suggestion
    enable_log_to_file: bool = Field(default=True)
```

File: .env.example (hunk 1 above)
Comment (dni) [reply .env.example:13]: hm, i feel like keeping it i string is more verbose, you could do `1 week`, or `3 days`.

https://loguru.readthedocs.io/en/stable/overview.html#easier-file-logging-with-rotation-retention-compression

Comment (motorina0) [reply .env.example:13]: - ok, I see. Please add some examples in the comment then.

**Review by motorina0 — APPROVED** [review]
LGTM 🥳

**Review by motorina0 — APPROVED** [review]

**Review by callebtc — APPROVED** [review]

**Pull request merged**